src/core/tsi/transport_security_interface.h

```diff
@@ -62,6 +62,15 @@ const char *tsi_result_to_string(tsi_result result);
 
 extern grpc_tracer_flag tsi_tracing_enabled;
 
+/* -- tsi_zero_copy_protector object --
+
+  This object protects and unprotects grpc slice buffers without memory copy
```

Comment:
We should be claiming that we *make it possible* for implementation to not make copies. For example, we're making one copy for integrity+privacy. TLS will make one copy as well.